contracts/NonfungiblePositionManager.sol

```diff
@@ -0,0 +1,144 @@
+    // function mint(MintParams calldata params) external payable returns (uint256 tokenId, BalanceDelta delta) {
+    //     (uint160 sqrtPriceX96,,,) = poolManager.getSlot0(params.range.key.toId());
+    //     (tokenId, delta) = mint(
+    //         params.range,
+    //         LiquidityAmounts.getLiquidityForAmounts(
+    //             sqrtPriceX96,
+    //             TickMath.getSqrtPriceAtTick(params.range.tickLower),
+    //             TickMath.getSqrtPriceAtTick(params.range.tickUpper),
+    //             params.amount0Desired,
+    //             params.amount1Desired
+    //         ),
+    //         params.deadline,
+    //         params.recipient,
+    //         params.hookData
+    //     );
+    //     require(params.amount0Min <= uint256(uint128(delta.amount0())), "INSUFFICIENT_AMOUNT0");
+    //     require(params.amount1Min <= uint256(uint128(delta.amount1())), "INSUFFICIENT_AMOUNT1");
+    // }
+
+    function increaseLiquidity(uint256 tokenId, uint256 liquidity, bytes calldata hookData, bool claims)
+        external
+        isAuthorizedForToken(tokenId)
+        returns (BalanceDelta delta)
+    {
+        delta = modifyLiquidity(tokenPositions[tokenId].range, liquidity.toInt256(), hookData, claims);
+    }
+
+    function decreaseLiquidity(uint256 tokenId, uint256 liquidity, bytes calldata hookData, bool claims)
+        public
+        isAuthorizedForToken(tokenId)
+        returns (BalanceDelta delta)
+    {
+        delta = modifyLiquidity(tokenPositions[tokenId].range, -(liquidity.toInt256()), hookData, claims);
+    }
+
+    function burn(uint256 tokenId, address recipient, bytes calldata hookData, bool claims)
+        external
+        isAuthorizedForToken(tokenId)
+        returns (BalanceDelta delta)
+    {
+        // remove liquidity
+        TokenPosition storage tokenPosition = tokenPositions[tokenId];
+        LiquidityRangeId rangeId = tokenPosition.range.toId();
+        Position storage position = positions[msg.sender][rangeId];
+        if (0 < position.liquidity) {
+            decreaseLiquidity(tokenId, position.liquidity, hookData, claims);
+        }
+        require(position.tokensOwed0 == 0 && position.tokensOwed1 == 0, "NOT_EMPTY");
+        delete positions[msg.sender][rangeId];
+        delete tokenPositions[tokenId];
+
+        // burn the token
+        _burn(tokenId);
+    }
+
+    // TODO: in v3, we can partially collect fees, but what was the usecase here?
+    function collect(uint256 tokenId, address recipient, bytes calldata hookData, bool claims)
+        external
+        returns (BalanceDelta delta)
+    {
+        delta = modifyLiquidity(tokenPositions[tokenId].range, 0, hookData, claims);
+    }
+
+    function feesOwed(uint256 tokenId) external view returns (uint256 token0Owed, uint256 token1Owed) {
+        TokenPosition memory tokenPosition = tokenPositions[tokenId];
+        return feesOwed(tokenPosition.owner, tokenPosition.range);
```

Comment:
question: why are we using the liquidity range as the fees owed for the lp token?